Source/WebCore/platform/graphics/skia/NativeImageSkia.cpp

```diff
@@ -62,8 +69,25 @@ Color NativeImage::singlePixelSolidColor() const
     if (size() != IntSize(1, 1))
         return Color();
 
-    notImplemented();
-    return Color();
+    auto platformImage = this->platformImage();
+    if (platformImage->isTextureBacked()) {
+        GrDirectContext* grContext = PlatformDisplay::sharedDisplayForCompositing().skiaGrContext();
+        auto* glContext = PlatformDisplay::sharedDisplayForCompositing().skiaGLContext();
+        GLContext::ScopedGLContextCurrent scopedCurrent(*glContext);
+        const auto& imageInfo = platformImage->imageInfo();
+        uint32_t pixel;
+        SkPixmap pixmap(imageInfo, &pixel, imageInfo.minRowBytes());
+        if (!platformImage->readPixels(grContext, pixmap, 0, 0))
+            return Color();
+
+        return pixmap.getColor(0, 0);
+    }
+
+    SkPixmap pixmap;
+    if (!platformImage->peekPixels(&pixmap))
+        return Color();
```

Comment:
Doesn't this do a full copy of the image? If it does, is it a performance concern for this patch?

Reply:
No, there aren't copies here, it just peek the pixels.

Reply:
Thanks for clarifying that (I can't resolve this conversation)